### ISSUE TITLE
[Major feature] Add integration for other email/calendar providers

### ISSUE THREAD

**Issue by MMachado05** [schej-it/schej.it#93]
I'm putting this here almost exclusively to mix the bag and throw in some new ideas, rather than properly requesting the feature; I'm sure implementing it would be a Goliathan undertaking.

Nonetheless, I *do* think it would be pretty cool if Schej had integration options with email providers beyond just Google; namely, something like Outlook would be great, as it's yet another popular email provider, but maybe even more niche, privacy-focused options, such as ProtonMail, or even NextCloud.

Obviously, integration for each one of these providers would probably require a bunch of code, and I'm sure there are more pressing features to worry about now, but it's something to consider should it ever be viable!

**Comment by jonyTF** [comment]
Hi, appreciate the feedback!

We are actually currently in the process of implementing Apple Calendar integration, and Outlook Calendar integration is after that on the roadmap. 

Quick question though: do you personally use outlook/some other calendar provider and is that what inspired this feature request?

**Comment by MMachado05** [comment]
I actually noticed the Apple integration post on Reddit _just_ after making this Issue!

Personally, I'm trying to switch to ProtonMail in order to lower my dependence on Google, primarily for philosophical reasons.

I'll be the first to tell you that it's likely the majority of people who [will] use Schej will be just fine with Google+Apple+Outlook integration; this would just allow more freedom for the edge cases that apply to more privacy-focused individuals like myself.

**Comment by jonyTF** [comment]
HAHA no way that's so funny

I see I see makes sense... currently thinking of an extensible way to architect the code to make it really easy to integrate more calendar providers in the future,,, so we may be seeing a ProtonMail integration after Outlook 👀

**Comment by tillhanke** [comment]
I just stumbled upon your project and the feature of importing events from external calendars is exactly what I was looking for! Its a absolute killer feature.
Would it easy to implement a simple ics link import to the provider based calendars? Because this would fully cover all non integrated calendars in one sweep ;) 
I personally use nextcloud, but there are probably also users, that might just use their email provider for calendar synching. To cover all these providers is of course senseless, but having the option of importing a ics link would solve this completely.

Thanks for your great work!

**Comment by avatar1024** [comment]
My sense is that it would be great to include at least one open source / free software provider, I'd personally be in favour of Nextcloud given it used by many individuals (myself included) and organisations.

And yes ics link import would be great as a universal solution.

**Comment by trey-stafford** [comment]
An ics link feature would be fantastic! My organization uses outlook, but there are restrictions on unverified apps preventing me from using the outlook integration.

**Comment by Igralino** [comment]
> preventing me from using the outlook integration.

Same thing for me! 
I agree that adding integrations with Outlook/iCloud/ics-link and manual .ics file import will be such a great features!

**Comment by ChrisLane** [comment]
> An ics link feature would be fantastic! My organization uses outlook, but there are restrictions on unverified apps preventing me from using the outlook integration.

Yep, ICS would likely be the most basic platform agnostic solution and may not need to introduce too much additional code/libraries. 
CalDAV would require fewer steps on the user's side (no need to find where to share an ICS URL) but would likely be a more advanced implementation.
